parachain-upgrade/src/lib.rs

```diff
@@ -81,17 +105,6 @@ decl_module! {
 		// TODO: figure out a better weight than this
 		#[weight = (0, DispatchClass::Operational)]
 		pub fn schedule_upgrade(origin, validation_function: Vec<u8>) {
-			ensure_root(origin)?;
-			System::<T>::can_set_code(&validation_function)?;
-			Self::schedule_upgrade_impl(validation_function)?;
-		}
-
-		/// Schedule a validation function upgrade without further checks.
-		///
-		/// Same as [`Module::schedule_upgrade`], but without checking that the new `validation_function`
-		/// is correct. This makes it more flexible, but also opens the door to easily brick the chain.
-		#[weight = (0, DispatchClass::Operational)]
-		pub fn schedule_upgrade_without_checks(origin, validation_function: Vec<u8>) {
```

Comment:
oh good q: there were no difference in the previous version. Yes, in `schedule_upgrade` if the upgrade is prohibited by polkadot, the code would bail from `can_set_code` if `code_upgrade_allowed` is none. In `schedule_upgrade_without_checks` we would go straight to `schedule_upgrade_impl` and it would still check `code_upgrade_allowed` and bail if it's none, which makes the behavior identical to that of `schedule_upgrade`.

That seemed poppycocky to me, so I removed it

Reply:
`can_set_code` is a Substrate function that checks that the given runtime is following some conventions, aka upgrading the spec_version etc. Please bring it back!

Reply:
ah, right, of course, I'll bring it back